internal/provider/ec2/ebs.go

```diff
@@ -846,7 +860,7 @@ func (v *ebsVolumeSource) attachOneVolume(
 	// Wait for the volume to move out of "creating".
 	volume, err := v.waitVolumeCreated(ctx, volumeId)
 	if err != nil {
-		return "", "", errors.Trace(maybeConvertCredentialError(err, ctx))
+		return "", "", errors.Trace(v.env.HandleCredentialError(ctx, err))
```

Comment:
I don't fully understand when we call `HandleCredentialError` and why.

The code changes above here imply to me that the pattern is we `HandleCredentialError` on the way out of the top level calls. 

So we don't call HandleCredentialError in the case of an error in `listVolumes`, but we do for `ListVolumes`.

However, there are cases like this which contradict my understanding

Reply:
> The code changes above here imply to me that the pattern is we HandleCredentialError on the way out of the top level calls.

That is the aim, but not necessarily the law. I suspect we will need to invest time to look at standardizing the providers, but that's not now, unfortunately.